test/integration/puppetlabs/puppetserver/error_handling_int_test.clj

```diff
@@ -56,4 +56,17 @@
               (is (= "Internal Server Error: java.lang.NullPointerException"
                      (:body response)))
               (is (re-matches #"text/plain; charset=.*"
-                              (get-in response [:headers "content-type"]))))))))))
+                              (get-in response [:headers "content-type"]))))))
+        (testing "Puppet 3 Master URLs are no longer valid"
+          (let [response (http-client/get
+                           "https://localhost:8140/production/catalog/localhost"
+                           bootstrap/request-options)]
+            (is (= 404 (:status response)))
+            (is (= (:body response) request-handler/error-message-404))))
+
+        (testing "Puppet 3 CA URLs are no longer valid"
+          (let [response (http-client/get
+                           "https://localhost:8140/production/certificate_status/localhost"
+                           bootstrap/request-options)]
+            (is (= 404 (:status response)))
+            (is (= (:body response) request-handler/error-message-404))))))))
```

Comment:
This test validates the behavior of the system returns a specific symbol, which is OK, but it would be a great improvement if the test validated the behavior the end user would expect.  In this situation, the end user should expect to see a helpful error message with server info, software and API versions, etc...

It'd be great to assert the string contains that information so if we change error-message-404 to something "bad" the tests will catch it for us.